### PR TITLE
Creation of the user lib

### LINKED ISSUE #2220: [User] Refactorize libs "account" & "auth" into "user"
The lib "account" has never been really clear.

Let's create a library "user" with two subfolder : 
- **auth**: _folder dedicated to the connected user_.
- **user**: _folder dedicated for any other user (like the members of an organization)_

## Auth
Let's keep the same structure as the current "auth" folder.
Add components from the current "account/profile" folder.

## User
Create a state with a CollectionService.
Add component from "organization/member".

### PR DESCRIPTION
**Pull request by Mussuranae**
Creation of the user lib, that means refactor of the auth and account lib. These two will be deleted and refactor in the user/auth lib and user/user lib.
Fix #2220

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -133,66 +65,24 @@
...
+      "@blockframes/contract/*": ["libs/contract/src/lib/*"],
+      "@blockframes/financial-report": ["libs/financial-report/src/index.ts"],
+      "@blockframes/financial-report/*": ["libs/financial-report/src/lib/*"],
+      "@blockframes/admin": ["libs/admin/src/index.ts"],
+      "@blockframes/admin/*": ["libs/admin/src/lib/*"],
+      "@blockframes/apps/delivery/*": ["apps/delivery/delivery/src/app/*"],
+      "@blockframes/apps/catalog/marketplace/*": ["apps/catalog/src/app/marketplace/*"],
+      "@blockframes/apps/catalog/dashboard/*": ["apps/catalog/src/app/dashboard/*"],
+      "@blockframes/apps/catalog/*": ["apps/catalog/catalog/src/app/*"],
+      "@blockframes/user/*": ["libs/user/src/lib/user/*"]
```
Comment (GrandSchtroumpf) [inline tsconfig.json]: Maybe create a link 
`"@blockframes/user": ["libs/user/src/lib/user/index.ts"]`

Comment (GrandSchtroumpf) [reply tsconfig.json]: Also to keep things organized could you put those aliases above the apps's one ?

File: libs/user/src/lib/user.module.spec.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { async, TestBed } from '@angular/core/testing';
+import { UserModule } from './user.module';
+
+describe('UserModule', () => {
```
Comment (GrandSchtroumpf) [inline libs/user/src/lib/user.module.spec.ts]: You can remove this file

File: apps/catalog/catalog/src/app/dashboard/layout/layout.component.html [hunk 3]
```diff
@@ -41,10 +41,10 @@
       </algolia-autocomplete>
       <div fxLayout="row" fxLayoutGap="24px">
         <overlay-notification-widget></overlay-notification-widget>
-        <account-profile-widget></account-profile-widget>
+        <auth-user-widget></auth-user-widget>
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/dashboard/layout/layout.component.html]: change to just `auth-widget` so we don't confuse auth & user

File: apps/catalog/catalog/src/app/dashboard/layout/layout.module.ts [hunk 4]
```diff
@@ -10,7 +10,7 @@ import { AlgoliaAutocompleteModule } from '@blockframes/ui/algolia-autocomplete/
 // Widgets
 import { NotificationWidgetModule } from '@blockframes/notification';
 import { SearchWidgetModule } from '@blockframes/ui/search-widget';
-import { ProfileWidgetModule } from '@blockframes/account/profile';
+import { UserWidgetModule } from '@blockframes/auth';
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/dashboard/layout/layout.module.ts]: rename `AuthWidgetModule`

File: libs/user/src/lib/auth/components/widget/widget.component.ts [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
-  selector: 'account-profile-widget',
-  templateUrl: './profile-widget.component.html',
-  styleUrls: ['./profile-widget.component.scss'],
+  selector: 'auth-user-widget',
+  templateUrl: './widget.component.html',
+  styleUrls: ['./widget.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class ProfileWidgetComponent {
+export class UserWidgetComponent {
```
Comment (GrandSchtroumpf) [inline libs/user/src/lib/auth/components/widget/widget.component.ts]: AuthWidgetComponent

File: styles/components/mat-button.scss [hunk 6]
```diff
@@ -72,7 +72,7 @@
   }
 
   // Add a border around the avatar of the user in the toolbar
-  account-profile-widget button[mat-icon-button] img {
+  auth-user-widget button[mat-icon-button] img {
```
Comment (GrandSchtroumpf) [inline styles/components/mat-button.scss]: Change here too.

**Comment by phpgeekfr** [comment]
Yes good job @Mussuranae ! But I noticed "user" functions in `auth.service.ts` ( the functions after `getOrCreateUserByMail` ) that should be in user state. Did you saw with @GrandSchtroumpf  if this was ok to let them in `auth.service.ts` for now ?

**Pull request merged**